app/assets/javascripts/projects/index.js

```diff
@@ -384,6 +439,218 @@
+      var index = $.inArray(rowId, selectedProjects);
+
+      // If checkbox is checked and row ID is not in list of selected project IDs
+      if (this.checked && index === -1) {
+        selectedProjects.push(rowId);
+      // Otherwise, if checkbox is not checked and ID is in list of selected IDs
+      } else if (!this.checked && index !== -1) {
+        selectedProjects.splice(index, 1);
+      }
+
+      updateDataTableSelectAllCtrl();
+      e.stopPropagation();
+    });
+
+    // Handle click on "Select all" control
+    $('.dataTables_scrollHead input[name="select_all"]').change(function(e) {
+      if (this.checked) {
+        $('.project-row-selector:not(:checked)').trigger('click');
+      } else {
+        $('.project-row-selector:checked').trigger('click');
+      }
+      // Prevent click event from propagating to parent
+      e.stopPropagation();
+    });
+  }
+
+  function updateSelectedRows() {
+    TABLE.rows().every(function() {
+      var rowSelector = $(this.node()).find('input[type="checkbox"]');
+      var rowId = this.data().DT_RowId;
+
+      if ($.inArray(rowId, selectedProjects) !== -1) {
+        rowSelector.prop('checked', true);
+      } else {
+        rowSelector.prop('checked', false);
+      }
+    });
+
+    updateDataTableSelectAllCtrl();
+  }
+
+  function dataTableInit() {
+    var TABLE_ID = '#projects-overview-table';
+    TABLE = $(TABLE_ID).DataTable({
+      dom: "R<'row'<'col-sm-9-custom toolbar'l><'col-sm-3-custom'f>>tpi",
+      stateSave: false,
+      processing: true,
+      serverSide: true,
+      scrollY: '64vh',
+      scrollCollapse: true,
+      destroy: true,
+      ajax: {
+        url: $(TABLE_ID).data('source'),
+        global: false,
+        type: 'POST',
+        data: function(params) {
+          params.filter = projectsViewFilter;
+          // return { ...params, ...{ filter: projectsViewFilter } };
+        }
+      },
+      colReorder: {
+        fixedColumnsLeft: 9
+      },
+      columnDefs: [{
+        // Checkbox column needs special handling
+        targets: 0,
+        searchable: false,
+        orderable: false,
+        className: 'dt-body-center',
+        sWidth: '1%',
+        render: function() {
+          return "<input class='project-row-selector' type='checkbox'>";
+        }
+      }, {
+        targets: 8,
+        searchable: false,
+        orderable: false,
+        className: 'dt-body-center',
+        sWidth: '1%'
+      }],
+      oLanguage: {
+        sSearch: I18n.t('general.filter')
+      },
+      rowCallback: function(row, data) {
+        // Get row ID
+        var rowId = data.DT_RowId;
+        var dropdown = $(row).find('.dropdown');
+        var dropdownCell = dropdown.closest('td');
+        // If row ID is in the list of selected row IDs
+        if ($.inArray(rowId, selectedProjects) !== -1) {
+          $(row).find('input[type="checkbox"]').prop('checked', true);
+        }
+
+        initEditProjectButton($(row));
```

Comment:
This `rowCallback` is triggered any time table is reloaded, e.g. on sorting of columns, filtering, etc. And in this function, we're initializing various event listeners - for edit & archive/restore functionalities. But these event listeners never get de-registered, do they? Could we somehow de-register them before (is there some "on table unload" callback) reloading the table?

Reply:
I think they should be removed together with DOM elements, they attached to, once content of the table is re-drawn.